internal/future/statestore/public_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Google LLC
```

Comment:
Rename to redis_test.go? I don't see any public.go stuff in here.

Reply:
This only tests the public interface methods - with passing in the cfg needed to create whatever is the backend of choice in createBackend. Except for createBackend, I expect the rest of the tests in this method to stay identical even if redis is swapped for some other backend.

redis_test IMO will have a check such as - CreateTicket validating that the backed redis db actually got the entry created (not via GetTicket - but via querying the db)

Reply:
This only tests the public interface methods - with passing in the cfg needed to create whatever is the backend of choice in createBackend. Except for createBackend, I expect the rest of the tests in this method to stay identical even if redis is swapped for some other backend.

redis_test IMO will have a check such as - CreateTicket validating that the backed redis db actually got the entry created (not via GetTicket - but via querying the db)